kernel/idle.c

```diff
@@ -14,9 +14,21 @@
 #include <zephyr/logging/log.h>
 #include <ksched.h>
 #include <kswap.h>
+#ifdef CONFIG_SHUTDOWN
+#include <zephyr/sys/shutdown_internal.h>
+#endif
 
 LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 
+#ifdef CONFIG_SHUTDOWN
+static bool shutdown_sheduled;
```

Comment:
Should this be an atomic? Maybe some weird cases on SMP or something, `z_sys_shutdown_schedule` could be called from wherever.

Reply:
Hmm... actually on SMP you'll have N idle threads, so I guess we should wait until all CPUs go to idle before shutting down. Please check latest revision.